src/synopsys-action/tools-parameter.ts

```diff
@@ -82,90 +83,109 @@ export class SynopsysToolsParameter {
-          if (fixPrSeverity != null && fixPrSeverity !== '') {
-            sarifReportFilterSeverities.push(fixPrSeverity.trim())
+    if (!isPrEvent) {
+      if (parseToBoolean(inputs.POLARIS_REPORTS_SARIF_CREATE)) {
+        /** Set Polaris SARIF inputs in case of non PR context */
+        const sarifReportFilterSeverities: string[] = []
+        const sarifReportFilterAssessmentIssuesType: string[] = []
+
+        if (inputs.POLARIS_REPORTS_SARIF_SEVERITIES) {
+          const filterSeverities = inputs.POLARIS_REPORTS_SARIF_SEVERITIES.split(',')
+          for (const sarifSeverity of filterSeverities) {
+            if (sarifSeverity) {
+              sarifReportFilterSeverities.push(sarifSeverity.trim())
+            }
           }
         }
-      }
 
-      if (inputs.POLARIS_REPORTS_SARIF_ISSUE_TYPES) {
-        const filterIssueTypes = inputs.POLARIS_REPORTS_SARIF_ISSUE_TYPES.split(',')
-        for (const issueType of filterIssueTypes) {
-          if (issueType != null && issueType !== '') {
-            sarifReportFilterAssessmentIssuesType.push(issueType.trim())
+        if (inputs.POLARIS_REPORTS_SARIF_ISSUE_TYPES) {
+          const filterIssueTypes = inputs.POLARIS_REPORTS_SARIF_ISSUE_TYPES.split(',')
+          for (const issueType of filterIssueTypes) {
+            if (issueType) {
+              sarifReportFilterAssessmentIssuesType.push(issueType.trim())
+            }
           }
         }
-      }
-      polData.data.polaris.reports = {
-        sarif: {
-          create: true,
-          severities: sarifReportFilterSeverities,
-          file: {
-            path: inputs.POLARIS_REPORTS_SARIF_FILE_PATH.trim()
-          },
-          issue: {
-            types: sarifReportFilterAssessmentIssuesType
-          },
-          groupSCAIssues: isBoolean(inputs.POLARIS_REPORTS_SARIF_GROUP_SCA_ISSUES) ? JSON.parse(inputs.POLARIS_REPORTS_SARIF_GROUP_SCA_ISSUES) : true
+        polData.data.polaris.reports = {
+          sarif: {
+            create: true,
+            severities: sarifReportFilterSeverities,
+            file: {
+              path: inputs.POLARIS_REPORTS_SARIF_FILE_PATH.trim()
+            },
+            issue: {
+              types: sarifReportFilterAssessmentIssuesType
+            },
+            groupSCAIssues: isBoolean(inputs.POLARIS_REPORTS_SARIF_GROUP_SCA_ISSUES) ? JSON.parse(inputs.POLARIS_REPORTS_SARIF_GROUP_SCA_ISSUES) : true
+          }
         }
       }
+      if (parseToBoolean(inputs.POLARIS_UPLOAD_SARIF_REPORT) && isNullOrEmptyValue(inputs.GITHUB_TOKEN)) {
+        /** Throw error if SARIF upload is enabled but GitHub token is empty */
+        throw new Error(constants.GITHUB_TOKEN_VALIDATION_SARIF_UPLOAD_ERROR)
+      }
+    } else {
+      if (parseToBoolean(inputs.POLARIS_REPORTS_SARIF_CREATE) && parseToBoolean(inputs.POLARIS_UPLOAD_SARIF_REPORT)) {
+        /** Log warning if SARIF create is enabled in PR context */
+        warning(constants.SARIF_REPORT_WARNING_FOR_PR_SCANS)
```

Comment:
I think both the parameters can be passed separately. '&&' should be converted to '||'

Reply:
updated, it was missed